web/src/components/storage/AddExistingDeviceMenu.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) [2025] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState } from "react";
+import { _, n_ } from "~/i18n";
+import { sprintf } from "sprintf-js";
+import {
+  Dropdown,
+  DropdownList,
+  DropdownItem,
+  DropdownGroup,
+  MenuToggleElement,
+  MenuToggle,
+  Divider,
+} from "@patternfly/react-core";
+import { MenuHeader } from "~/components/core";
+import MenuDeviceDescription from "./MenuDeviceDescription";
+import { useAvailableDevices, useConfigModel, useModel } from "~/queries/storage";
+import { deviceLabel } from "~/components/storage/utils";
+
+export default function AddExistingDeviceMenu() {
+  const [isOpen, setIsOpen] = useState(false);
+  const toggle = () => setIsOpen(!isOpen);
+  const allDevices = useAvailableDevices();
+  const model = useConfigModel({ suspense: true });
+  const modelHook = useModel();
+
+  const drivesNames = model.drives.map((d) => d.name);
+  const devices = allDevices.filter((d) => !drivesNames.includes(d.name));
+
+  const Header = ({ drives }) => {
+    const desc = sprintf(
+      n_(
+        "Extends the installation beyond the currently selected disk",
+        "Extends the installation beyond the current %d disks",
+        drives.length,
+      ),
+      drives.length,
+    );
+
+    return <MenuHeader title={_("Select another disk to define partitions")} description={desc} />;
+  };
+
+  if (!devices.length) return null;
+
+  return (
+    <Dropdown
+      isOpen={isOpen}
+      onOpenChange={toggle}
+      onSelect={toggle}
+      onActionClick={toggle}
+      toggle={(toggleRef: React.Ref<MenuToggleElement>) => (
+        <MenuToggle
+          ref={toggleRef}
+          onClick={toggle}
+          aria-label={_("Use additional disk toggle")}
+          isExpanded={isOpen}
+        >
+          {_("Use additional disk")}
+        </MenuToggle>
+      )}
+    >
+      <DropdownList>
+        {/* @ts-ignore:next-line See https://github.com/patternfly/patternfly/issues/7327 */}
```

Comment:
Please, use @ts-expect-error instead. It will let us know when there is no an error anymore if PatternFly finally ends up changing the label prop type. In such case, we can drop the directive instead of leaving there forever, which might hide another typing error in the future.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments

Reply:
And the same with @ts-ignore

> Hey-o 👋 I might seem to be helpful, but you should really avoid using me in the code!